### PR TITLE
Added some tests to prevent regressions in partitioning and flexattention

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #137065
* #137049
* #137043
* __->__ #136826

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: test/inductor/test_flex_attention.py [hunk 1]
```diff
@@ -1671,6 +1673,65 @@ def mask_mod(b, h, q, kv):
         out = func(query, key, value, block_mask=block_mask)
         out.sum().backward()
 
+    @supported_platform
+    def test_compiled_block_mask(self):
```
Comment (yanboliang) [inline test/inductor/test_flex_attention.py:1677]: should we move this under ```class TestBlockMask```? since it's a block mask specific test?

**Review by yanboliang — APPROVED** [review]

File: test/inductor/test_flex_attention.py (hunk 1 above)
Comment (Chillee) [reply test/inductor/test_flex_attention.py:1677]: Oh, I didn't know we had that section lol

File: test/inductor/test_perf.py [hunk 2]
```diff
@@ -695,6 +695,13 @@ def f(a, b):
         inp = (T(10, grad=True), T(10, grad=True))
         self.assertExpectedInline(count_numel_train(f, *inp), """70""")
 
+    def test_partitioning_relu(self):
```
Comment (drisspg) [inline test/inductor/test_perf.py:698]: is this related?

**Review by drisspg — APPROVED** [review]

File: test/inductor/test_perf.py (hunk 2 above)
Comment (Chillee) [reply test/inductor/test_perf.py:698]: No
